### PR TITLE
Fixed zeromq compilation and version check problem

### PR DESCRIPTION
**Pull request by RickAi**

### CODE REVIEW

File: cmake/modules/zeromq.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@
 ### ZeroMQ ###
 
 # TODO(legend): check zeromq version?
-find_path(ZMQ_INCLUDE_DIR NAMES zmq.hpp PATHS ${ZMQ_SEARCH_PATH})
+find_path(ZMQ_INCLUDE_DIR NAMES zmq.h* PATHS ${ZMQ_SEARCH_PATH}) # sometime it appears as .hpp or .h
```
Comment (kygx-legend) [inline cmake/modules/zeromq.cmake]: We do not requrie `zmq.h` but `zmq.hpp`

File: cmake/modules/zeromq.cmake [hunk 2]
```diff
@@ -35,14 +53,16 @@ else(ZMQ_FOUND)
         ExternalProject_Add(
             cppzmq
             GIT_REPOSITORY "https://github.com/zeromq/cppzmq"
+            GIT_TAG "v4.2.1"
             PREFIX ${THIRDPARTY_DIR}
             UPDATE_COMMAND ""
             CONFIGURE_COMMAND ""
             BUILD_COMMAND ""
-            INSTALL_COMMAND cp ${THIRDPARTY_DIR}/src/cppzmq/zmq.hpp ${PROJECT_BINARY_DIR}/include/zmq.hpp
+            INSTALL_COMMAND mkdir -p ${PROJECT_BINARY_DIR}/include && cp ${THIRDPARTY_DIR}/src/cppzmq/zmq.hpp ${PROJECT_BINARY_DIR}/include/zmq.hpp
```
Comment (kygx-legend) [inline cmake/modules/zeromq.cmake]: Not necessary.

**Comment by RickAi** [comment]
@kygx-legend @Yuzhen11 
The existing build error on MacOs can be fixed with this change:
https://github.com/RickAi/flexps/commit/ce4559bd254645040120d590b4f807b09684b840
I am not sure whether it's necessary to send a pull request, if need please tell me.

**Comment by kygx-legend** [comment]
yes, please go ahead.

**Pull request closed without merging**